lib/storage/index_db.go

```diff
@@ -440,6 +440,16 @@ func marshalMetricIDs(dst []byte, metricIDs []uint64) []byte {
 	//
 	// The srcBuf is a []byte cast of metricIDs.
 	srcBuf := unsafe.Slice((*byte)(unsafe.Pointer(unsafe.SliceData(metricIDs))), 8*len(metricIDs))
+	if len(srcBuf) == 0 {
+		// If the list of metricIDs is empty, then add one byte so that zstd has
+		// something to compress. Otherwise nothing will be added to dst and if
+		// dst is empty the record won't be added to the cache and the search
+		// for a given filter will be performed again and again. This may lead
+		// to cases like this:
+		// https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7009
+		srcBuf = []byte{0}
```

Comment:
There is no need in compressing `srcBuf` in this case. This will save CPU time needed for calling `encoding.CompressZSTDLevel()`.

Reply:
Done.